### PR TITLE
Introduce cloud-speech API config yaml.

### PR DESCRIPTION
**Pull request by jmuk**
- changed RecognizeNonStreaming() to idempotent

"Recognize" method is a bidi-streaming method, which is not
supported by Gapic codegen, but yet it should exist when
we want to support it for some language.

Anyways streaming methods will be ignored as
https://github.com/googleapis/toolkit/pull/328

### CODE REVIEW

File: google/cloud/speech/v1/cloud_speech_gapic.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+type: com.google.api.codegen.ConfigProto
+generate_samples: true
+language_settings:
+  java:
+    package_name: com.google.cloud.speech.v1
```
Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Java currently puts vkit under spi.{MAJOR_VERSION} (I realized I forgot to mention this in your other PR)

File: google/cloud/speech/v1/cloud_speech_gapic.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    max_retry_delay_millis: 60000
+    initial_rpc_timeout_millis: 60000
+    rpc_timeout_multiplier: 1
+    max_rpc_timeout_millis: 60000
+    total_timeout_millis: 600000
+  methods:
+  - name: Recognize
+    retry_codes_name: non_idempotent
+    retry_params_name: default
+    timeout_millis: 30000
```
Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: use 60000

File: google/cloud/speech/v1/cloud_speech_gapic.yaml (hunk 1 above)
Comment (jmuk) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Done. I'll create a fix for language API -- but I'd want to say this is a bug of configgen (cc: @shinfan)

Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: I fixed config gen yesterday: https://github.com/googleapis/toolkit/commit/9652cb012458e299e2077ad67e5472cba0dca2b3

File: google/cloud/speech/v1/cloud_speech_gapic.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+type: com.google.api.codegen.ConfigProto
+generate_samples: true
+language_settings:
+  java:
+    package_name: com.google.cloud.speech.spi.v1
```
Comment (shinfan) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: @garrettjonesgoogle  Do we still need spi here?

Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Yes, there hasn't been a decision to stop using it yet.

File: google/cloud/speech/v1/cloud_speech_gapic.yaml (hunk 2 above)
Comment (bjwatson) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Is this something that should be updated in the autogen tool for the GAPIC config? (FYI @ethanbao)

Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Already done in https://github.com/googleapis/toolkit/commit/9652cb012458e299e2077ad67e5472cba0dca2b3

Comment (bjwatson) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Cool cool

**Comment by jmuk** [comment]
So, is this good to go?

**Comment by garrettjonesgoogle** [comment]
You still need to change the non-retry timeouts to 60 seconds.

**Comment by jmuk** [comment]
Rebased, and fixed the timeout, also fixed nodejs config for the integration with gcloud-node. PTAL.

File: google/cloud/speech/v1/cloud_speech_gapic.yaml [hunk 4]
```diff
@@ -13,6 +13,8 @@ language_settings:
     package_name: Google::Cloud::Speech::V1
   php:
     package_name: Google\Cloud\Speech\V1
+  nodejs:
+    package_name: "@google-cloud/speech"
```
Comment (garrettjonesgoogle) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: Could you file an issue to add support for the generating the node package in the config generator?

**Comment by garrettjonesgoogle** [comment]
LGTM

File: google/cloud/speech/v1/cloud_speech_gapic.yaml (hunk 4 above)
Comment (jmuk) [inline google/cloud/speech/v1/cloud_speech_gapic.yaml]: There's a PR https://github.com/googleapis/toolkit/pull/351 to use this.

**Pull request merged**